Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabFocusHandler.cpp

```diff
@@ -436,60 +416,62 @@ namespace AzToolsFramework::Prefab
         }
     }
 
-    void PrefabFocusHandler::OpenInstanceContainers(const AZStd::vector<AZ::EntityId>& instances) const
+    void PrefabFocusHandler::SetInstanceContainersOpenState(const AZStd::vector<RootAliasPath>& instances, bool openState) const
     {
         // If this is called outside the Editor, this interface won't be initialized.
         if (!m_containerEntityInterface)
         {
             return;
         }
-        
-        for (const AZ::EntityId& containerEntityId : instances)
+
+        for (const RootAliasPath& rootAliasPath : instances)
         {
-            InstanceOptionalReference instance = GetReferenceFromContainerEntityId(containerEntityId);
+            InstanceOptionalReference instance = GetInstanceReferenceFromRootAliasPath(rootAliasPath);
```

Comment:
I think there is potential for an optimization here. If you are traversing the same hierarchy multiple times, you already know that a path is valid and don't need to keep on checking FindNestedInstance() yields a valid instance or not for every path. I guess it depends on the order of paths in the vector whether you traverse the same hierarchy multiple times or not.

Another reason why benchmarks would be nice to have so that we can decide whether an optimization is truly needed or not.

Reply:
Yeah I mentioned it in the description :) I can optimize this function, and also the one that generates the path for the breadcrumb. I just didn't want to rewrite the whole class in a single, big PR so I'm doing that as a second step.

Agreed on the benchmark!